File: Exercism/java/resistor-color/src/main/java/ResistorColor.java
class ResistorColor {
    int colorCode(String color) {
        String[] colors = colors();
        int n = colors.length;
        for (int i = 0; i < n; i++) {
            if (colors[i].equals(color))
                return i;
        }
        return -1;
    }

    String[] colors() {
        return new String[] {"black", "brown", "red", "orange",
         "yellow", "green", "blue", "violet", "grey", "white"};
    }
}

File: Exercism/c/resistor-color/src/resistor_color.c
#include "resistor_color.h"

resistor_band_t color_code(resistor_band_t band) {
    return band;
}

const resistor_band_t *colors() {
    return bands;
}

const resistor_band_t bands[] = {BLACK, BROWN, RED, ORANGE, YELLOW,
    GREEN, BLUE, VIOLET, GREY, WHITE};